controllers/vspheredeploymentzone_controller.go

```diff
@@ -156,8 +156,7 @@ func (r vsphereDeploymentZoneReconciler) reconcileNormal(deploymentZoneCtx *capv
 	if err := r.Client.Get(deploymentZoneCtx, failureDomainKey, failureDomain); err != nil {
```

Comment:
Happy to drop the entire second commit from this PR, but it makes sense in the context of the first IMO and works toward solving the context issue here. 

I'm also happy to get https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2369 merged first and then rebase.

Reply:
Reviewed and approved 2369.